bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/form/ReservedProperties.java

```diff
@@ -0,0 +1,171 @@
+    public static final String PN_VALIDATION_EXPRESSION_MESSAGE = "validateExpMessage"; // reusing the same property name as in foundation
+    public static final String PN_MULTISELECT = "multiSelect";
+    public static final String PN_MULTISELECTION = "multiSelection";
+    public static final String PN_ENABLE_UNCHECKED_VALUE = "enableUncheckedValue";
+    public static final String PN_CHECKED_VALUE = "checkedValue";
+    public static final String PN_UNCHECKED_VALUE = "uncheckedValue";
+    public static final String PN_MAX_FILE_SIZE = "maxFileSize";
+    public static final String PN_FILE_ACCEPT = "accept";
+    public static final String PN_BUTTON_TEXT = "buttonText";
+    public static final String PN_WRAP_DATA = "wrapData";
+    public static final String PN_FRAGMENT_PATH = "fragmentPath";
+    public static final String PN_BUTTON_TYPE = "buttonType";
+    public static final String PN_THANK_YOU_MSG_V1 = "thankyouMessage";
+    public static final String PN_THANK_YOU_MSG_V2 = "thankYouMessage";
+    public static final String PN_THANK_YOU_OPTION = "thankYouOption";
+    public static final String PN_RUNTIME_DOCUMENT_PATH = FormContainer.PN_RUNTIME_DOCUMENT_PATH;
+    public static final String PN_CLOUD_SERVICE_PATH = "cloudServicePath";
+    public static final String PN_RECAPTCHA_CLOUD_SERVICE_PATH = "rcCloudServicePath";
+    public static final String PN_RECAPTCHA_SIZE = "recaptchaSize";
+    public static final String PN_BREAK_BEFORE_TEXT = "breakBeforeText";
+    public static final String PN_BREAK_AFTER_TEXT = "breakAfterText";
+    public static final String PN_OVERFLOW_TEXT = "overflowText";
+    public static final String PN_ALT_TEXT = "altText";
+    public static final String PN_IMAGE_SRC = "imageSrc";
+    public static final String PN_FILE_REF = "fileReference";
+    public static final String PN_SHOW_APPROVAL_OPTION = "showApprovalOption";
+    public static final String PN_SHOW_LINK = "showLink";
+    public static final String PN_SHOW_AS_POPUP = "showAsPopup";
+    public static final String PN_TEXT_IS_RICH = "textIsRich";
+    public static final String PN_MULTILINE = "multiLine";
+    public static final String PN_DESIGN_DEFAULT_TYPE = Title.PN_DESIGN_DEFAULT_TYPE;
+    public static final String PN_TITLE_LINK_DISABLED = Title.PN_TITLE_LINK_DISABLED;
+    public static final String PN_DRAG_DROP_TEXT = "dragDropText";
+    public static final String PN_CLIENTLIB_REF = "clientLibRef";
+    public static final String PN_REDIRECT = "redirect";
+    public static final String PN_PREFILL_SERVICE = "prefillService";
+    public static final String PN_SPEC_VERSION = "specVersion";
+    public static final String PN_RICH_TEXT = "richText";
+
+    public static List<String> getNodeProperties() {
+        List<String> nodeProperties = new ArrayList<>();
+        Field[] fields = ReservedProperties.class.getDeclaredFields();
+
+        for (Field field : fields) {
```

Comment:
Can we store the properties in a singleton class so that the below loop with reflection doesn't get executed every single time

Reply:
+1. lets have a static field here to store the props.